src/scriptsys/script.h

```diff
@@ -0,0 +1,7 @@
+#pragma once
+
+#include "lua/lua.h"
```

Comment:
Thanks for the PR!
It would be nice for compile times and decoupling if we could forward declare the types to interface with the script system instead of including lua.h.

To forward declare a struct in C:
```typedef struct lua_State lua_State;```

In C, static means 'this symbol is local to this compile unit', which is a bit confusing compared to static members of a class.
You will want to put ```static lua_State* ms_luastate;``` in the .c file only, at the top.
You could make an accessor for the lua state if needed, or declare it as a global by marking it extern in the header.
I would discourage the global route though.

Generally the order for things in a C file are like so:
1. includes
2. defines, macros
3. enums and constants
4. type declarations
5. static function declarations (the prototypes)
5. global (public) and static (private) variables (might want them split for neatness)
6. global (public) function definitions (implementation)
7. static (private) function definitions (implementation)

This order ensures all the declarations are made before the definitions use them.